### PR TITLE
Zipper - RGWUser

### PR DESCRIPTION
**Pull request by dang**
Signed-off-by: Daniel Gryniewicz <dang@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dang** [comment]
Teuthology run is good:
http://pulpito.ceph.com/dang-2019-12-16_10:58:58-rgw-wip-dang-zipper-4-distro-basic-smithi/

**Comment by dang** [comment]
Ping @cbodley

**Comment by dang** [comment]
Hmm... Rebase messed up somehow.

File: src/rgw/rgw_auth.cc [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include "rgw_user.h"
 #include "rgw_http_client.h"
 #include "rgw_keystone.h"
+#include "rgw_sal.h"
```
Comment (mattbenjamin) [inline src/rgw/rgw_auth.cc]: still wish we could rename this

Comment (mattbenjamin) [reply src/rgw/rgw_auth.cc]: and it's namespace component

**Comment by dang** [comment]
jenkins retest this please

**Pull request merged**